src/gui/shell.cpp

```diff
@@ -1321,6 +1343,9 @@ void Shell::showEvent(QShowEvent* ev)
 	if (m_nvim->isReady()) {
 		init();
 	}
+
+	screenChanged();
+	connect(window()->windowHandle(), &QWindow::screenChanged, this, &Shell::screenChanged);
```

Comment:
Because showEvent can happen for other reasons besides show() we can

- call connect with Qt::UniqueConnection as third argument
- wrap this in a check for QShowEvent::spontaneous().

Just being pedantic here, but check window() for NULL.

Reply:
Added NULL check for window.